packages/kbn-apm-synthtrace-client/src/lib/otel/index.ts

```diff
@@ -0,0 +1,210 @@
+    schema_url?: string;
+  };
+  scope?: {
+    attributes?: {
+      'service.framework.name'?: string;
+      'service.framework.version'?: string;
+    };
+    dropped_attributes_count?: number;
+    name?: string;
+  };
+  name?: string;
+  trace_id?: string;
+  trace?: { id: string };
+  span_id?: string;
+  span?: { id: string };
+  dropped_attributes_count?: number;
+  dropped_events_count?: number;
+  dropped_links_count?: number;
+  timestamp_us?: number;
+}
+
+class Otel extends Serializable<OtelDocument> {
+  constructor(fields: OtelDocument) {
+    super({
+      ...fields,
+    });
+  }
+
+  error(spanId: string) {
+    return new OtelError({
+      ...this.fields,
+      attributes: {
+        'exception.message': 'boom',
+        'exception.type': '*errors.errorString',
+        'error.stack_trace': 'Error: INTERNAL: Boom',
+        'processor.event': 'error',
+        'timestamp.us': 1726580752010657,
+        'event.name': 'exception',
+      },
+      data_stream: {
+        dataset: 'generic.otel',
+        namespace: 'default',
+        type: 'logs',
+      },
+      'event.name': 'exception',
+      dropped_attributes_count: 0,
+      resource: {
+        attributes: {
+          'agent.name': 'opentelemetry/go',
+          'agent.version': '1.28.0',
+          'service.name': 'sendotlp-synth',
+          'service.instance.id': '89117ac1-0dbf-4488-9e17-4c2c3b76943a',
+          'some.resource.attribute': 'resource.attr',
+        },
+        dropped_attributes_count: 0,
+        schema_url: 'https://opentelemetry.io/schemas/1.26.0',
+      },
+      scope: {
+        attributes: {
+          'service.framework.name': 'sendotlp-synth',
+          'service.framework.version': '',
+        },
+        dropped_attributes_count: 0,
+        name: 'sendotlp-synth',
+      },
+      span_id: spanId,
+    });
+  }
+
+  metric() {
+    return new OtelMetric({
+      ...this.fields,
+      attributes: {
+        'metricset.name': 'service_destination',
+        'processor.event': 'metric',
+        'event.outcome': 'success',
+        'service.target.name': 'foo_service',
+        'service.target.type': 'http',
+        'span.name': 'child1',
+      },
+      data_stream: {
+        dataset: 'service_destination.10m.otel',
+        namespace: 'default',
+        type: 'metrics',
+      },
+      metrics: {
+        service_summary: 2,
+      },
+      resource: {
+        attributes: {
+          'agent.name': 'otlp',
+          'agent.version': '1.28.0',
+          'service.instance.id': '89117ac1-0dbf-4488-9e17-4c2c3b76943a',
+          'service.name': 'sendotlp-synth',
+          'metricset.interval': '10m',
+        },
+        dropped_attributes_count: 0,
+      },
+      scope: {
+        dropped_attributes_count: 0,
+        name: 'github.com/elastic/opentelemetry-collector-components/connector/spanmetricsconnectorv2',
+      },
+    });
+  }
+
+  transaction(id: string) {
+    return new OtelTransaction({
```

Comment:
Just a minor comment on the naming `OtelTransaction`: In OTel there is no such thing as transaction - there are only spans. We have transactions in our data model.

What we do is that in the enrichment lib, we just look for root spans and we enrich those with these `transaction.*` attributes, because those root spans are then the equivalent of our transactions.

Totally up to you how you call this, just wanted add this. Maybe `OtelRootSpan` or something like that would be better fit, but `OtelTransaction` is fine as well.

Reply:
Thank you for pointing that out and explaining! I added a comment to explain that but I would prefer to keep it as a transaction because that's what we use in APM and it will match the same naming so it will be easier to read, also we used `transaction` everywhere already.